mitodl/ocw-studio
# Log info when tables disappear

#### Pre-Flight checklist


#### What are the relevant tickets?
#1478 

#### What's this PR do?
This PR adds functionality to log a more informative sentry error when html -> md conversion in CKEditor drops tables. We've actually found the source of this bug (#1486) I think that this PR still adds some useful stuff that is worth keeping and could help debug future markdown strangeness. 

Some highlights:
- Adds a `validateHtml2md` function that runs some sanity checks on the html -> md conversion.
- Swaps out `@sentry/browser` for `@sentry/react`, and puts `SiteContentForm` inside a sentry's [Error Boundary](https://reactjs.org/docs/error-boundaries.html) component. If an error occurs, rather than the whole site crashing, we have controllable behavior.
- Adds [`jest-watch-typeahead`](https://www.npmjs.com/package/jest-watch-typeahead) for developer sanity.

#### How should this be manually tested?
1. Set up sentry environment variables:

    ```
    SENTRY_DSN=https://ba06ef0080e342db9b9ae1db53944f1a@o48788.ingest.sentry.io/5739976
    OCW_STUDIO_ENVIRONMENT=your-name-local
    ```
    
    We use `OCW_STUDIO_ENVIRONMENT` as the sentry environment. So by setting it to something specific to you, the sentry errors that you generate will be put in a separate bucket. 
2. Restart docker.
3. Trigger the table error (#1486), but tldr:
    1. Create a page with a table.
    2. Reload studio, and start creating a new Resource List. (You do not need to actually create it, just load the form.)
    3. Close the resource list and navigate to your table page
    4. You should see an error like the screenshots below.
4. Check that your entry was logged to sentry at (https://sentry.io/organizations/mit-office-of-digital-learning/issues/?environment=cac-local&project=5739976&query=url%3A%22http%3A%2F%2Fcac.ocw-studio.odl.local%3A8043%2Fsites%2Fchris-another-site%2Ftype%2Fpage%2Fedit%2F8e3640f4-95ce-4e26-ba47-197553c9da00%2F%22; switch the environment to yours).
    - the error should include the html and markdown
5. Remove the sentry stuff from your .env file to avoid logging extra errors to our sentry, and restart docker again.

#### Screenshots (if appropriate)

<img width="400" alt="Screen Shot 2022-09-20 at 9 37 49 AM" src="https://user-images.githubusercontent.com/9010790/191272909-c04b91f0-4688-43b6-99a5-dde77f656ba6.png">
